src/content/app/entity-viewer/variant-view/population-allele-frequencies/PopulationAlleleFrequencies.tsx

```diff
@@ -49,11 +55,37 @@ const PopulationAlleleFrequencies = (props: Props) => {
   });
 
   useEffect(() => {
-    if (currentData?.populationGroups && !populationGroup) {
-      setPopulationGroup(currentData.populationGroups[0]);
+    const groups = currentData?.populationGroups;
+
+    if (
+      groups?.length &&
+      Object.keys(populationFreqAccordionState).length === 0
+    ) {
+      const initialState = Object.fromEntries(
+        groups.map((group, index) => [group, index === 0])
+      );
+      setPopulationFreqAccordionState(initialState);
     }
   }, [currentData?.populationGroups]);
```

Comment:
The current react guidelines _really_ want people to reduce the use of `useEffect` only to cases where it's absolutely necessary. For example there is a linter rule (which we haven't enabled) that will complain if useEffect is used to set the state.

I am sure there are a lot of places in our codebase where we use an effect for that; but I think it's time we start listening to what react guidelines are saying.

I'm about to push a commit that rearranges the code a bit and removes the `useEffect`.


------

Incidentally, I think there was also a problem that collapsed sections persisted their collapsed state when the allele got changed. This has been fixed as a part of code rearrangement:

https://github.com/user-attachments/assets/bc9c219d-e57e-49b2-bd96-defeac96e23e